integrated_channels/blackboard/models.py

```diff
@@ -111,6 +111,112 @@ def get_learner_data_transmitter(self):
         return BlackboardLearnerTransmitter(self)
 
 
+@python_2_unicode_compatible
+class BlackboardLearnerAssessmentDataTransmissionAudit(models.Model):
+    """
+    The payload correlated to a courses subsection learner data we send to blackboard at a given point in time for an
+    enterprise course enrollment.
+
+    .. no_pii:
+    """
+    blackboard_user_email = models.CharField(
+        max_length=255,
+        blank=False,
+        null=False
+    )
+
+    enterprise_course_enrollment_id = models.PositiveIntegerField(
+        blank=False,
+        null=False,
+        db_index=True
+    )
+
+    course_id = models.CharField(
+        max_length=255,
+        blank=False,
+        null=False,
+        help_text="The course run's key which is used to uniquely identify the course for blackboard."
+    )
+
+    subsection_id = models.CharField(
+        max_length=255,
+        blank=False,
```

Comment:
are we *always* going to send subsections?

I ask because subsection_id and subsection_name both have `blank=False` which would fail validation if they aren't included, but based on the rest of the logic, we seem to allow for the prospect of not having a subsection grade all the time.

Reply:
Yikes, yeah we're not always going to be sending subsections, this is shared between course completion and assessment level reporting. Idk what I was thinking...

Reply:
or.... are they shared? Hold on, I think I confused myself. Investigating.

Reply:
Ok sorry- my brain was thinking of the base `LearnerDataTransmissionAudit`. It being "shared" as in it's the base 🤦, and the base has the integration ID and name as optional.

Reply:
Having said that, both the subsection ID and subsection name are a part of the subsection grade data retrieved from platform's gradebook API and that is the only place where we create these audits. So I guess we _**should**_ be always sending those pieces of data with this particular audit?

Reply:
ah yeah I didn't pay close attention to the class name (I missed "Assessment" in it). Yeah given the only reason this would be checked against/run is for assessment reporting then yes I think it's quite reasonable to expect subsection id and name.